packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -347,6 +354,9 @@ export class EditableText extends AbstractPureComponent2<EditableTextProps, IEdi
         /* eslint-disable-next-line deprecation/deprecation */
         const { altKey, ctrlKey, metaKey, shiftKey, which } = event;
         if (which === Keys.ESCAPE) {
+            if (this.props.stopPropagation === true) {
```

Comment:
please add a default value of false for the new prop in `public static defaultProps`. then you can simplify this boolean check:

```suggestion
            if (this.props.stopPropagation) {
```